File: app/src/main/java/com/example/newsweatherapp/Rest/ApiClient.java
package com.example.newsweatherapp.Rest;


import android.util.Log;


import java.util.concurrent.TimeUnit;

import okhttp3.logging.HttpLoggingInterceptor;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class ApiClient {

    public static final String Domain_Live="https://api.spaceflightnewsapi.net/v4/";


    public static final String BASE_URL = Domain_Live;

    private static Retrofit retrofit = null;


    public static Retrofit getResponse() {
        final okhttp3.OkHttpClient okHttpClient = new okhttp3.OkHttpClient.Builder()
                .readTimeout(120, TimeUnit.SECONDS)
                .connectTimeout(120, TimeUnit.SECONDS)
                .build();

        HttpLoggingInterceptor logging = new HttpLoggingInterceptor();
        logging.setLevel(HttpLoggingInterceptor.Level.BASIC);

        if (retrofit == null) {

            retrofit = new Retrofit.Builder()
                    .baseUrl(BASE_URL)
                    .client(okHttpClient)
                    .addConverterFactory(GsonConverterFactory.create())
                    .build();

            Log.e("ctdctftygfyhuihjio",""+retrofit.baseUrl());
        }
        return retrofit;
    }




}

File: app/src/main/java/com/example/newsweatherapp/Rest/ApiClient.kt
package com.example.newsweatherapp.Rest

import android.accessibilityservice.GestureDescription
import android.util.Log


import retrofit2.Retrofit
import okhttp3.OkHttpClient
import okhttp3.logging.HttpLoggingInterceptor
import retrofit2.converter.gson.GsonConverterFactory
import java.util.concurrent.TimeUnit





object ApiClient {
    const val Domain_Live = "https://api.spaceflightnewsapi.net/v4/"
    const val BASE_URL = Domain_Live
    private var retrofit: Retrofit? = null
    val response: Retrofit?
        get() {
            val okHttpClient: OkHttpClient = OkHttpClient.Builder()
                .readTimeout(120, TimeUnit.SECONDS)
                .connectTimeout(120, TimeUnit.SECONDS)
                .build()
            val logging = HttpLoggingInterceptor()
            logging.level = HttpLoggingInterceptor.Level.BASIC
            if (retrofit == null) {
                retrofit = Retrofit.Builder()
                    .baseUrl(BASE_URL)
                    .client(okHttpClient)
                    .addConverterFactory(GsonConverterFactory.create())
                    .build()

            }
            return retrofit
        }
}